Source/SwiftLintFramework/Reporters/HTMLReporter.swift

```diff
@@ -30,144 +30,82 @@ public struct HTMLReporter: Reporter {
-            "\t\t\t\t\t\t<b>Severity</b>\n",
-            "\t\t\t\t\t</th>\n",
-            "\t\t\t\t\t<th style=\"width: 500pt;\">\n",
-            "\t\t\t\t\t\t<b>Message</b>\n",
-            "\t\t\t\t\t</th>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t</thead>\n",
-            "\t\t\t<tbody>\n", rows, "\t\t\t</tbody>\n",
-            "\t\t</table>\n",
-            "\t\t\n",
-            "\t\t<br/>\n",
-            "\t\t\n",
-            "\t\t<h2>Summary</h2>\n",
-            "\t\t\n",
-            "\t\t<table>\n",
-            "\t\t\t<tbody>\n",
-            "\t\t\t\t<tr>\n",
-            "\t\t\t\t\t<td>Total files with violations</td>\n",
-            "\t\t\t\t\t<td>\(fileCount)</td>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t\t<tr>\n",
-            "\t\t\t\t\t<td>Total warnings</td>\n",
-            "\t\t\t\t\t<td>\(warningCount)</td>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t\t<tr>\n",
-            "\t\t\t\t\t<td>Total errors</td>\n",
-            "\t\t\t\t\t<td>\(errorCount)</td>\n",
-            "\t\t\t\t</tr>\n",
-            "\t\t\t</tbody>\n",
-            "\t\t</table>\n",
-            "\t\t\n",
-            "\t\t<hr />\n",
-            "\t\t\n",
-            "\t\t<p>\n",
-            "\t\t\tCreated with\n",
-            "\t\t\t<a href=\"https://github.com/realm/SwiftLint\"><b>SwiftLint</b></a>\n",
-            "\t\t\t", swiftlintVersion, " on ", dateString, "\n",
-            "\t\t</p>\n",
-            "\t</body>\n",
-            "</html>"
-        ].joined()
+        let parameters = [
+            "VIOLATIONS": rows,
+            "TOTAL_VIOLATING_FILES": String(fileCount),
+            "TOTAL_WARNINGS": String(warningCount),
+            "TOTAL_ERRORS": String(errorCount),
+            "VERSION": swiftlintVersion,
+            "DATE": dateString
+        ]
+
+        return generateReport(parameters: parameters)
+    }
+
+    private static func generateReport(parameters: [String: String]) -> String {
+        var report = loadTemplate()
+        for (key, value) in parameters {
+            report = report.replacingOccurrences(of: "$$\(key)$$", with: value)
+        }
+
+        return report
+    }
+
+    private static func loadTemplate() -> String {
+        let path = templatePath()
+
+        #if os(Linux)
+            // swiftlint:disable:next force_try
+            let data = try! Data(contentsOf: URL(fileURLWithPath: path))
+            return String(data: data, encoding: .utf8)!
+        #else
+            // swiftlint:disable:next force_try
+            return try! String(contentsOfFile: path)
+        #endif
+    }
+
+    private static func templatePath() -> String {
+        #if SWIFT_PACKAGE
+            let components = Array(NSString(string: #file).pathComponents.dropLast() + ["Templates", "template.html"])
+            let path = components.joined(separator: "/")
+        #else
+            // swiftlint:disable:next nesting
```

Comment:
Thought process:

"Getting bundles on Linux can't be broken! We use it to print the version and that surely works"

[goes to confirm]

```
.build/debug/swiftlint version
fatal error: unexpectedly found nil while unwrapping an Optional value
Current stack trace:
0    libswiftCore.so                    0x00007fc4f9554fc0 swift_reportError + 117
1    libswiftCore.so                    0x00007fc4f9567df0 _swift_stdlib_reportFatalError + 63
2    libswiftCore.so                    0x00007fc4f93756a3 <unavailable> + 0
3    libswiftCore.so                    0x00007fc4f94d2dfd <unavailable> + 0
4    libswiftCore.so                    0x00007fc4f93756a3 <unavailable> + 0
5    libswiftCore.so                    0x00007fc4f94946b0 specialized _fatalErrorMessage(StaticString, StaticString, StaticString, UInt, flags : UInt32) -> Never + 96
6    swiftlint                          0x000000000042c086 <unavailable> + 0
7    libpthread.so.0                    0x00007fc4f858cae9 <unavailable> + 0
8    libswiftCore.so                    0x00007fc4f9565b20 swift_once + 90
9    swiftlint                          0x000000000042c16a <unavailable> + 0
10   swiftlint                          0x000000000042c1e9 <unavailable> + 0
11   swiftlint                          0x000000000042c6b6 <unavailable> + 0
12   swiftlint                          0x0000000000447285 <unavailable> + 0
13   swiftlint                          0x000000000044b0c1 <unavailable> + 0
14   swiftlint                          0x000000000044865f <unavailable> + 0
15   swiftlint                          0x0000000000448fc4 <unavailable> + 0
16   swiftlint                          0x0000000000448ac1 <unavailable> + 0
17   swiftlint                          0x0000000000416b91 <unavailable> + 0
18   swiftlint                          0x0000000000416cd7 <unavailable> + 0
19   libdispatch.so                     0x00007fc4f974c2a7 <unavailable> + 0
20   libdispatch.so                     0x00007fc4f975b152 <unavailable> + 0
21   libdispatch.so                     0x00007fc4f978361f <unavailable> + 0
22   libpthread.so.0                    0x00007fc4f858570a <unavailable> + 0
23   libc.so.6                          0x00007fc4f73de7c0 clone + 109
Illegal instruction
```

💥